deel/influenciae/benchmark/influence_factory.py

```diff
@@ -202,9 +202,7 @@ def build(self, training_dataset: tf.data.Dataset, model: tf.keras.Model,
             dataset_hessian = training_dataset
         else:
             batch_size = training_dataset._batch_size.numpy()  # pylint: disable=W0212
-            take_size = int(
-                np.ceil(float(self.dataset_hessian_size) / batch_size)) * batch_size
-            dataset_hessian = training_dataset.take(take_size)
+            dataset_hessian = training_dataset.unbatch().take(self.dataset_hessian_size).batch(batch_size)
```

Comment:
If we have to unbatch it, isn't it better to ask the user to provide a dataset that is not batched in the first place?

Reply:
We could do that, but the problem is that these factories are being used inside `InfluenceCalculator`s, where the datasets are actually asked to be batched.
The change here is that when applying the `take` method, it is no longer recognized as a batched dataset and doesn't pass some asserts further in the code (inside RPS LJE)

Reply:
Ok, I guess it is alright for now but we should be careful of user feedback if it is causing trouble!